src/bin/cargo-fmt.rs

```diff
@@ -178,9 +178,9 @@ pub enum WorkspaceHitlist {
 }
 
 impl WorkspaceHitlist {
-    pub fn get_some<'a>(&'a self) -> Option<&'a [String]> {
-        if let &WorkspaceHitlist::Some(ref hitlist) = self {
-            Some(&hitlist)
+    pub fn get_some(&self) -> Option<&[String]> {
+        if let WorkspaceHitlist::Some(ref hitlist) = *self {
```

Comment:
I'm not sure why Clippy suggests this change - it looks fine either way to me

Reply:
I think the idea here is to prefer not to repeat `&` in pattern:

```rust
match *self {
    A => ...,
    B => ...,
    C => ...,
}
```
instead of
```rust
match self {
    &A => ...,
    &B => ...,
    &C => ...,
}
```
I agree that it is irrelevant in the case of `if let Some(..) = *self` v.s. `if let &Some(..) = self`.
Just to keep it consistent, I would like to use the former style here as well, if you don't mind.

Reply:
Sure, I don't mind either way